### PR TITLE
feat: support django 4.2 new STORAGES format

### LINKED ISSUE #716: Support Django 4.2 - settings.DEFAULT_FILE_STORAGE is deprecated
`settings.DEFAULT_FILE_STORAGE` is deprecated in Django 4.2 alpha. Use `STORAGES` instead.

### PR DESCRIPTION
**Pull request by bellini666**
Fix #716

### CODE REVIEW

**Comment by claudep** [comment]
Thanks! Would you also like to suggest a PR which adds Django 4.2 in the testing matrix?

**Comment by bellini666** [comment]
> Thanks! Would you also like to suggest a PR which adds Django 4.2 in the testing matrix?

done! I updated test.yaml and tox.ini, not sure if it is enough

File: sorl/thumbnail/conf/defaults.py [hunk 1]
```diff
@@ -1,42 +1,46 @@
 from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
 
 # When True ThumbnailNode.render can raise errors
 THUMBNAIL_DEBUG = False
 
 # Backend
-THUMBNAIL_BACKEND = 'sorl.thumbnail.base.ThumbnailBackend'
+THUMBNAIL_BACKEND = "sorl.thumbnail.base.ThumbnailBackend"
```
Comment (claudep) [inline sorl/thumbnail/conf/defaults.py:8]: Please avoid mixing this sort of changes which are not directly related to the commit. If you find it's important, you could suggest a different PR with those quote changes.

Comment (bellini666) [reply sorl/thumbnail/conf/defaults.py:8]: Sure, I didn't even notice my editor auto formatted the code. Sorry! :(

**Comment by claudep** [comment]
Any clue about the test failures?

**Comment by bellini666** [comment]
> Any clue about the test failures?

@claudep looking at the tests, they failed with `ModuleNotFoundError: No module named 'django'`.

Strange that I did not modify any behaviour there, I just added python 3.11 and django 4.2 to the test matrix.

I made one typo which I [just pushed a fix](https://github.com/jazzband/sorl-thumbnail/pull/728/commits/769b249407d834edec68c134238dd94bf1b3550f), but I don't believe it would be that.

Either way, can you approve the tests to run again?

**Review by claudep — APPROVED** [review]
Looks good now!

By the way, I think we should move the `THUMBNAIL_STORAGE` setting to a key from the `STORAGES` dict, so we can get the storage directly from the `django.core.files.storage.storages` structure. Would you like to continue working on this?

**Comment by bellini666** [comment]
> By the way, I think we should move the THUMBNAIL_STORAGE setting to a key from the STORAGES dict, so we can get the storage directly from the django.core.files.storage.storages structure. Would you like to continue working on this?

Sure!Can you open an issue for that and assign it to me, so that I don't forget it? =P
